.github/workflows/csharp.yml

```diff
@@ -52,35 +56,40 @@ jobs:
       EVENT_TYPE: ${{ github.event_name }}
       CURRENT_BRANCH: ${{ github.head_ref || github.ref_name }}
     steps:
-      - uses: actions/checkout@v3
-      - name: Setup .NET
-        uses: actions/setup-dotnet@v1
-        with:
-          dotnet-version: 5.0.x
-      - name: Restore nuget packages
-        run: |
-          nuget restore OptimizelySDK.Travis.sln
-          nuget install ./OptimizelySDK.Tests/packages.config -OutputDirectory ./packages
-          nuget install NUnit.Runners -Version 2.6.4 -OutputDirectory ./testrunner
-      - name: script
-        run: |
-          ./install_mono.sh
-          xbuild /p:SignAssembly=true /p:AssemblyOriginatorKeyFile=$(pwd)/keypair.snk /p:Configuration=Release ./OptimizelySDK.Travis.sln
-          mono ./testrunner/NUnit.Runners.2.6.4/tools/nunit-console.exe ./OptimizelySDK.Tests/bin/Release/OptimizelySDK.Tests.dll
-      - name: find and sign dll
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1
+      - name: Setup NuGet
+        uses: NuGet/setup-nuget@v1
+      - name: Restore NuGet packages
+        run: nuget restore ./OptimizelySDK.Travis.sln
+      - name: Build solution
+        run: msbuild /p:SignAssembly=true /p:AssemblyOriginatorKeyFile=$(pwd)/keypair.snk /p:Configuration=Release ./OptimizelySDK.Travis.sln
+      - name: Install NUnit Console
```

Comment:
Please move it before running msbuild. SO all setups can be seen together.

Reply:
Sure. That makes sense.  

I intended to only execute installations before they were needed in case of failure.  eg If build failed, no need to install NUnit.

Reply:
🤔 Thinking more on this… I'd like to think green and code sustainably here.  

By performing the installation only _if_ and just before it's needed, we're reducing the amount of resources (albeit small) being used on a potentially unnecessary step.  

If you feel strongly over this re-ordering, please DM me on Teams, and I'll roll it into another PR.